packages/lesswrong/components/hooks/useRecordPostView.tsx

```diff
@@ -112,8 +120,25 @@ export const useRecordPostView = (post: ViewablePost) => {
       console.log("recordPostView error:", error); // eslint-disable-line
     }
   }, [postsRead, setPostRead, increasePostViewCount, sendVertexViewItemEvent, currentUser, recordEvent]);
+
+  const recordPostCommentsView = ({ post }: Pick<RecordPostViewArgs, 'post'>) => {
+    if (currentUser) {
+      if (!postsRead[post._id]) {
+        // Update the client-side read status cache.
+        // Set the value to true even if technically we might be saving isRead: false on the server, if the post hasn't been read before, to get the correct UI update.
+        setPostRead(post._id, true);
+
+        // Only send update to server the first time a user interacts with the comments after the page is rendered (and the client-side cache isn't populated yet).
+        // Otherwise it becomes much more likely that someone else posts a comment after the first time we send this,
+        // but then we send it again because e.g. the user clicked on another comment (from the initial render).
```

Comment:
I'm not seeing the logic you have that achieves this.

Reply:
It being in the if-block at all depends on `postsRead` not having been populated for that post id yet, which happens on L129 (and so happens the first time `recordPostCommentsView` is called).  I'll try to clarify the comment